### DISCUSSION TITLE
FCM 테스트 이용에 대해서

### DISCUSSION THREAD

**Discussion by wocjf0513** [SCBJ-7/SCBJ-BE#193] [Announcements]
> FCM TEST를 위해서 Thymeleaf등 테스트를 할 수 있는 환경을 조성했습니다.

관련 파일들이 
- FCMController 
```java
@Controller
@RequestMapping("/v1/fcm")
public class FCMController {

    @GetMapping
    public String fcmTestPage() {
        return "FcmTest";
    }
}

```
- FCMRestController
```java
@RestController
@RequiredArgsConstructor
@ResponseStatus(HttpStatus.OK)
@RequestMapping("/v1/fcm")
public class FCMRestController {

    private final FCMService fcmService;

    @PostMapping
    public ResponseDTO<String> saveFCMToken(@RequestParam(name = "email") String email, @RequestParam(name = "token") String token) {
        fcmService.saveToken(email, token);
        return ResponseDTO.res("토큰 저장에 성공했습니다.");
    }

    @PostMapping("/alarm")
    public ResponseDTO<String> pushAlarm(@RequestParam(name = "email") String email) {
        fcmService.sendMessageTo(email, new Data("TEST용 알림", "TEST용 푸쉬 알림이 생성되었습니다.", LocalDateTime.now()));
        return ResponseDTO.res("푸쉬 알림에 성공했습니다.");
    }
}
```
이 있고요.

## 테스트 서순
1. localhost:8080/v1/fcm 을 GET으로 호출 시키면 TEST용 페이지를 발견합니다.
2. 개발자 콘솔로 들어가면 기기별 FCM TOKEN이 있습니다.
3. 이를 localhost:8080/v1/fcm POST로 email과 fcm token을 request body로 요청합니다.
4. 그러면, fcm token이 서버에 저장되고,
5. localhost:8080/v1/fcm/alarm?email=wocjf0513 다음과 같이 아까 요청으로 보낸 email을 POST로 요청하면
6. 콘솔에 푸쉬 알림으로 온 데이터가 보이게 됩니다.

5을 제외하고 자신이 만든 API에서 알림 관련 API를 사용 시한다면, 
자신의 API 호출 URL 로 요청하면
6 콘솔에 푸쉬 알림으로 온 데이터가 보이게 됩니다.!!

## [필요 설정](https://www.notion.so/Back-End-493832eceb0e46fe9f0cd3132b8d8abd?p=c8680a636c054628acf5610e239ed58c&pm=s)

**Comment by Kwonminwoo** [comment]
와우 감사합니다!!
복잡하군요 ㅋㅋㅋㅋ

**Reply by wocjf0513** [reply]
ㅋㅋㅋㅋ하다가 이해안되시거나 안되는 부분있으면 말씀주셔요!!
